packages/twenty-server/src/engine/core-modules/record-crud/services/upsert-record.service.ts

```diff
@@ -73,16 +75,29 @@ export class UpsertRecordService {
         );
       }
 
+      const restrictedFields =
+        repository.objectRecordsPermissions?.[
+          objectMetadataItemWithFieldsMaps.id
+        ]?.restrictedFields;
+
       const objectRecordWithFilteredFields = Object.keys(objectRecord).reduce(
-        (acc, key) => {
-          if (fieldsToUpdateArray.includes(key)) {
-            return {
-              ...acc,
-              [key]: objectRecord[key],
-            };
+        (acc, fieldName) => {
+          if (!fieldsToUpdateArray.includes(fieldName)) {
+            return acc;
+          }
+
```

Comment:
I feel like we should have solved the permission issue in generateFindToolInputSchema and not here, no? Here you kind of make the AI think that it's possible but then it's not. We should just not expose what is not possible?

Reply:
I'm not sure it's worth keeping it here, is it a good idea to swallow errors?